### PR TITLE
Add codegen for indexing references on the right-hand side.

### PR DESCRIPTION
**Pull request by vext01**
This is one half of the changes we talked about yesterday.

Another PR needs to follow to cover indexing a ref on the left-hand side.

### CODE REVIEW

File: ykcompile/src/lib.rs [hunk 1]
```diff
@@ -1993,4 +2018,25 @@ mod tests {
         ct.execute(&mut args);
         assert_eq!(args.1, 4);
     }
+
+    /// Test codegen of indexing a struct ref on the right-hand side.
```
Comment (vext01) [inline ykcompile/src/lib.rs:2022]: Just realized that this isn't "indexing".

**Comment by vext01** [comment]
Try that.

**Comment by ptersilie** [comment]
Looks good. Please squash.

**Comment by vext01** [comment]
splat.

**Pull request merged**